Create resource

lionants02 opened the pull request.

piruin committed:
Add LICENSE

piruin committed:
Rename project

piruin committed:
Add .travis.yml

piruin committed:
Set version by git's tag

piruin committed:
Setup kotlinter

piruin committed:
Add postman

lionants02 committed:
add pgsql and redis dao.

lionants02 committed:
code format.

lionants02 committed:
move dao package.

lionants02 committed:
add log system and clean dependencies block.

lionants02 committed:
add unit test dependencies and log config.

lionants02 committed:
add mockito to dependency.

lionants02 committed:
add AccessTokenResource with dummy token manager.

lionants02 committed:
Rename TokenManager to AccessTOkenManager

lionants02 committed:
add security role

lionants02 committed:
add comment to RoleTokenSecurity

lionants02 committed:
add session resource

lionants02 committed:
re-code

lionants02 committed:
fix SessionsManager

lionants02 committed:
add jwt access token

lionants02 committed:
Rename TokenDao to ApiKeyDao

lionants02 committed:
add DemoApiKeyDao to AccessTokenResource

lionants02 committed:
Update security filter.

lionants02 committed:
update security jwt test.

lionants02 committed:
update build gradle add kotlinx-coroutines

lionants02 committed:
update auth
* add PgSqlApiKeyDao
* add InMemoryApiKeyDao

lionants02 committed:
move dao and add session resource

lionants02 committed:
Update session.
- add anonymous create session
- add InMemoryRegisterStore

lionants02 committed:
add fail test InMemoryRegisterStoreDao.

lionants02 committed:
add PgSqlRegisterStoreDao

lionants02 committed:
add fail test in JwtSessionsManagerTest

lionants02 committed:
update jwt verify.

lionants02 committed:
reflactor dao package struct.

lionants02 committed:
add get by session add register storage and clean parameter pgSql

lionants02 committed:
add timestamp in create api key and register session.

lionants02 committed:
add vital

lionants02 committed:
add Hikari connection pool.

lionants02 committed:
fix bug build.gradle.kts

lionants02 committed:
add HII_ALONE for stan alone server.

lionants02 committed:
Add get result and convert datetime.

lionants02 committed:
update .gitignore add gradle wrapper.

lionants02 committed:
update to gradlew 5.4.1

lionants02 committed:
fix bug update to gradlew 5.4.1
ถ้าไม่มี gradle config จะมีปัญหาเวลา build เพราะ ระบบจะใช้ gradle ในเครื่องซึ่งมักจะเป็น version เก่า

lionants02 committed:
add docker build

lionants02 committed:
Merge branch 'master' into CreateResource

lionants02 committed:
Add error response filter and get log level.

lionants02 committed:
fix endpoint token.

lionants02 committed:
fix bug update sessions.

lionants02 committed:
fix bug vital resource ถ้าบน class ไม่มี @Path resource rest จะไม่ถูกโหลด

lionants02 committed:
update error message เวลาไม่พบ API

lionants02 committed:
add readme

lionants02 committed:
fix bug build jar library

lionants02 committed:
fix bug hikari sql driver.

lionants02 committed:
fix create apikey template

lionants02 committed:
add redis key store.

lionants02 committed:
Optimize pgSql unit test 16 min to 1 min

lionants02 committed:
fix permission gradlew

lionants02 committed:
git ignore include gradle-wrapper.properties
ไม่งั้นจะ build บน travis ไม่ได้ travis require gradle-wrapper.properties

lionants02 committed:
add gradle-wrapper.properties

lionants02 committed:
Update docker build.

lionants02 committed:
สร้าง JwtPrincipal เองเพื่อเวลาใช้จะได้ไม่ต้อง get access token แต่สามารถเข้าถึง object jwt ได้เลย

lionants02 committed:
update build.gradle.kts

lionants02 committed:
csrf fix

lionants02 committed:
add gson java.time SERIALIZER.

lionants02 committed:
fix access token expire 30 min.

lionants02 committed:
Jwt token expire return 403.

lionants02 committed:
fix bug java time on docker.

lionants02 committed:
getResult เวลาดึงค่าที่ไม่ได้วัดจะ null แก้ถ้าค่าไหนไม่ได้วัดจะ null

lionants02 committed:
fix Asia/Bangkok timezone

lionants02 committed:
Update postman

lionants02 committed:
rename SqlDevice to SqlApiKeyStore

lionants02 committed:
Sql object to internal object.

lionants02 committed:
add last result dao

lionants02 committed:
clean pgSqlTestRule

lionants02 committed:
set last result when get result.

piruin commented on src/main/kotlin/hii/thing/api/sessions/SessionsManager.kt:
```diff
@@ -17,10 +17,12 @@
 
+import java.security.Principal
+
 interface SessionsManager {
```
ส่วน DAO รับ Token เหมือนเดิมจะดีกว่าครับ เพราะการรับ Principal แล้ว .name เพื่อเอา Token ทำให้ JwtSessionManager ผูกติดกับ JwtPrincipal ไปเลย

piruin commented on src/main/kotlin/hii/thing/api/sessions/SessionsResource.kt:
```diff
@@ -42,26 +42,26 @@ class SessionsResource(
-        val session = sessionsManager.getBy(accessToken)
-        val create = sessionsManager.updateCreate(accessToken, detail)
+        val userPrincipal = context.userPrincipal
+        val session = sessionsManager.getBy(userPrincipal)
```
cast` UserPrincipal` เป็น `JwtPrincipal` แล้ว `sessionsManager.getBy(userPrincipal.accessToken)`

lionants02 committed:
change input type SessionsManager is accessToken

lionants02 committed:
add blood level

lionants02 committed:
Verify jwt ไม่ผ่าน ส่ง 401

lionants02 committed:
return 404
ถ้า request ข้อมูลที่จำเป็นต้อง Authen แต่ไม่มีการระบุ header Authorization : Bearer

lionants02 committed:
Optimize log.

lionants02 committed:
jwt expire return 401

lionants02 committed:
optimize docker build.

lionants02 committed:
fix jar error "Please add log4j-core to the classpath."

lionants02 committed:
role kios to kiosk

lionants02 committed:
update token not found return 401

lionants02 committed:
add rsa string key pair dao.

lionants02 committed:
add KeyPairManage.setUp(dao)
-ใช้สำหรับเปลี่ยนที่จัดเก็บ key

lionants02 committed:
change get private and public key from system env

lionants02 committed:
add JwtApiKeyDao ใช้สำหรับ base token เป็นแบบ JWT

lionants02 committed:
เปลี่ยน last_result ให้เก็บ result ล่าสุดไปเลย

lionants02 committed:
เพิ่ม name ใน CreateSessionDetail ตาม postman

lionants02 committed:
fix demo in memory api key change role kios to kiosk

lionants02 committed:
เวลา new session จะส่งค่าเก่าที่เคยวัดตาม input type.
โดยขอบเขตการส่งคืนข้อมูลขึ้นอยู่กับ input type.

lionants02 committed:
update last result add value ref link.

lionants02 committed:
change PgSqlApiKeyDao to JwtApiKeyDao

lionants02 committed:
ให้ระบบใส่ deviceId ให้อัตโนมัตติ โดยดึงจาก jwt

lionants02 committed:
ไม่ตรวจสอบชื่อ หากเสียบบัตร

lionants02 committed:
add ref link in last result.

lionants02 committed:
ดึง token จาก query token ได้

lionants02 committed:
fix error pg unit test.

lionants02 committed:
add verify path with jwt scope.

lionants02 committed:
add ThingPrincipal and rename scopt to scope

lionants02 committed:
add Jwt create link

lionants02 committed:
add kdoc

lionants02 committed:
rename refLink to tokenLink

lionants02 committed:
ดึง Result ที่เคยวัดไปครั้งที่แล้ว
?token=xxxx

lionants02 committed:
เพิ่มตัวสร้าง device token

lionants02 committed:
add some kdoc

lionants02 committed:
some fix private scope.

lionants02 committed:
dokka config javadoc

lionants02 committed:
add config log4j2 output logstash.

lionants02 committed:
fix bug null โดยใส่ subject ใน token
เพราะ ในส่วน security มีการ log subject ไว้ด้วย

lionants02 committed:
Rename and add replay link.
- rename refLink to replayLink.
- add return link /result?token=xxx

lionants02 committed:
Add kdoc.

lionants02 committed:
Rename ApiKeyDao to DeviceTokenDao

lionants02 committed:
Enable check scope.

lionants02 committed:
fix bug jwt create access token scope

lionants02 committed:
แสดง Error เวลา JWT Verify ไม่ผ่าน

lionants02 committed:
update CreateDevice and update PgSqlTestRule

lionants02 committed:
add call api log.

lionants02 committed:
log4j plugin ใน logstash ยกเลิกใช้งาน
แนะให้เปลี่ยนมาใช้ filebeat แทน

lionants02 committed:
rename log to logger

lionants02 committed:
reflector Sessions Resource

lionants02 committed:
คำนวนอายุด้วยวันเกิดที่ใส่เข้ามาตอนขอ Session

lionants02 committed:
Critical fix device token
- คำนวนวันหมดอายุผิดจาก 10 ปี เป็น 10 วัน

lionants02 committed:
add log การวัด

lionants02 committed:
Device Token not expire

lionants02 committed:
add session log

lionants02 committed:
fix bug ไม่กรอกเลขบัตรประชาชน

lionants02 committed:
add sex เวลาขอ session

lionants02 committed:
fix bug เวลาไม่ใส่เลขบัตรประชาชน

lionants02 committed:
add blood pressures level

lionants02 committed:
add track sex and input type

lionants02 committed:
add sex

lionants02 committed:
fix bug update session.

lionants02 committed:
replay link ใส่ url เต็ม

lionants02 committed:
add enum sex

lionants02 committed:
ตอนขอ Session ใหม่ ให้เอา replay link ออก

lionants02 committed:
add BPLevel

lionants02 committed:
update unit test sex.

lionants02 committed:
fix format

lionants02 committed:
add cache e-tag

lionants02 committed:
update log

lionants02 committed:
create jetty log

lionants02 committed:
update jetty log

lionants02 committed:
init flyway

lionants02 committed:
Add sex and Suggestions sex ต้องใช้ตัวใหญ่ MALE, FEMALE

lionants02 committed:
kotlin format

lionants02 committed:
fix bug Ci

lionants02 committed:
add DemoRSAKeyPairDao สำหรับทดสอบ และ standalone

lionants02 committed:
change DaoConfig InMemory to DemoRSAKeyPairDao สำหรับ standalone

lionants02 committed:
add BloodPressures suggestions

lionants02 committed:
fix blood pressures unit test

lionants02 committed:
Update README.md

lionants02 committed:
sex to upper

lionants02 committed:
add undefined sex

lionants02 committed:
สถาพยาบาล to สถานพยาบาล

lionants02 committed:
replayLink to shareableLink

lionants02 committed:
ดึงผลย้อนหลัง ไม่ต้องแสดง shareable link อีก fix report role

lionants02 committed:
ถ้าใส่ citizen id แล้ว จะไม่สามารถ update citizen id ซ้ำได้

lionants02 committed:
enable info build

lionants02 committed:
JwtCont class to object

lionants02 committed:
add JwtTestRule

lionants02 committed:
remove require test spec in gradle.build

lionants02 committed:
fix JwtTestRule for local test.

lionants02 committed:
เพิ่ม InputType เป็นแบบ enum

lionants02 committed:
add HistoryBloodPressuresDao

lionants02 committed:
HistoryBloodPressuresDao add unit test check time

lionants02 committed:
เพิ่มชุด ทดสอบพื้นฐานสำหรับ History

lionants02 committed:
add check time test util

lionants02 committed:
add check time `should be equal to`

lionants02 committed:
add HistoryHeightsDao

lionants02 committed:
add HistoryWeightDao

lionants02 committed:
เรียงลำดับ ล่าสุดขึ้นก่อน

lionants02 committed:
remove old history

lionants02 committed:
rename JwtTestRule to TestRule

lionants02 committed:
rename TestRule to InMemoryTestRule

lionants02 committed:
add new HeightHistory

lionants02 committed:
Add class History

lionants02 committed:
เพิ่มอายุ ตอนเสียบบัตรลง Report

lionants02 committed:
เพิ่ม pulse ใน bp

lionants02 committed:
เพิ่ม history

lionants02 committed:
ในกรณีไม่ได้ใส่เพศมา จะไปดึงมาจากของเดิม

lionants02 committed:
fix error last sex

lionants02 committed:
edit name

lionants02 committed:
rename class Device to DeviceToken

lionants02 committed:
reflector package

lionants02 committed:
เปลี่ยนชื่อให้ตรงกับหน้าที่ class
- DeviceToken to DeviceKey

lionants02 committed:
fix bug ตอนที่จะดึงข้อมูลจากคีย์ ลืมใส่ตัว ID มาด้วย

lionants02 committed:
เพิ่ม fun ให้สามารถดึงข้อมูลการสร้าง DeviceKey จาก accessToken ได้

lionants02 committed:
create device token by ธ.ก.ส. group

lionants02 committed:
add group nsho

lionants02 committed:
move keyPair to parameter

lionants02 committed:
update pg sql test

lionants02 committed:
add DeviceDao

lionants02 committed:
add DeviceResource

lionants02 committed:
เพิ่มการ link table session กับ device และ ทั่วไป
- ทำให้ ตาราง session เชื่อม device id กับ ตาราง device
- เวลาดึงค่า env ให้ลองดึงค่าจาก Property ก่อน ถ้าไม่เจอค่อยไปดึงค่าจาก env

lionants02 committed:
add device id index and add foreign key

lionants02 committed:
add all foreign key

lionants02 committed:
ปรับ unit test ให้รองรับ foreign key

lionants02 committed:
ตอนขอ session ให้เก็บ ip address ด้วย

lionants02 committed:
PSQLExceptionFilter

lionants02 committed:
fix bug allow device

lionants02 committed:
add real ip in log

lionants02 committed:
fix bug เวลาขอ session

lionants02 committed:
update postman

lionants02 committed:
update postman

lionants02 committed:
remove bug postman

lionants02 committed:
fix bug genkey

lionants02 committed:
time first column

lionants02 committed:
เพิ่ม Device name

lionants02 committed:
fix bug device id

lionants02 committed:
ย้าย device id ออกจาก access token

lionants02 committed:
add Master key

lionants02 committed:
update git ignore

lionants02 committed:
update Postman

lionants02 committed:
add error filter.

lionants02 committed:
update postman

lionants02 committed:
update unit test

lionants02 committed:
update git ignore

lionants02 committed:
fix rsa key remove new line

lionants02 committed:
Update ThingPrincipal
- add get type
- add get device id
- add get device location

lionants02 committed:
add postman NHSO-hdfs

lionants02 committed:
ลบ fun ซ้ำซ้อนใน ThingPrincipal

lionants02 committed:
Ignore CreateMasterKey.kt

lionants02 committed:
add SendNHSO

lionants02 committed:
fix no roll

lionants02 committed:
update master token expire 1 day.

lionants02 committed:
update postman

lionants02 committed:
update postman remove test env

lionants02 committed:
update postman history endpoint
